Fix backend API certificate generation

priteau opened the pull request:
Some dependencies for generating backend API certificates were missing but were undetected because they were fulfilled by other vault playbooks run earlier during the deployment process.

This was detected when playbook hosts were extended to include separate network hosts running some OpenStack services.

markgoddard commented on etc/kayobe/ansible/vault-generate-backend-tls.yml:
```diff
@@ -1,10 +1,42 @@
 ---
+# Required for uri module to work with self-signed certificates and for systems to trust
+# the self-signed CA
+- name: Install CA on network hosts
```
Network hosts? Target is controllers

markgoddard replied:
Maybe we need to expand the target hosts in vault-deploy-overcloud.yml?

priteau replied:
> Network hosts? Target is controllers

My mistake, I forgot to edit when I removed `network` hosts from my changes.

priteau replied:
> Maybe we need to expand the target hosts in vault-deploy-overcloud.yml?

I have a preference for having explicit dependencies, otherwise we can run into issues when nodes are reprovisioned for example.

priteau committed:
Fix backend API certificate generation

Some dependencies for generating backend API certificates were missing
but were undetected because they were fulfilled by other vault playbooks
run earlier during the deployment process.

This was detected when playbook hosts were extended to include separate
network hosts running some OpenStack services.
